dispatcher/DAQController.py

```diff
@@ -89,161 +87,90 @@ def SolveProblem(self, latest_status, goal_state):
-        active_states = [STATUS.ARMING, STATUS.ARMED, STATUS.RUNNING,
-                         STATUS.ERROR, STATUS.UNKNOWN]
-        if goal_state['tpc']['active'] == 'false':
-
-            # Send stop command if we have to
-            if (
-                    # TPC not in Idle, error, timeout
-                    (latest_status['tpc']['status'] in active_states) or
-                    # MV linked and not in Idle, error, timeout
-                    (latest_status['muon_veto']['status']  in active_states and
-                     goal_state['tpc']['link_mv'] == 'true') or
-                    # NV linked and not in Idle, error, timeout
-                    (latest_status['neutron_veto']['status']  in active_states and
-                     goal_state['tpc']['link_nv'] == 'true')
-            ):
-                self.StopDetectorGently(detector='tpc')
-            elif latest_status['tpc']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('tpc')
-
-        # 1b - deal with MV but only if MV not linked to TPC
-        if goal_state['tpc']['link_mv'] == 'false' and goal_state['muon_veto']['active'] == 'false':
-            if latest_status['muon_veto']['status'] in active_states:
-                self.StopDetectorGently(detector='muon_veto')
-            elif latest_status['muon_veto']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('muon_veto')
-        # 1c - deal with NV but only if NV not linked to TPC
-        if goal_state['tpc']['link_nv'] == 'false' and goal_state['neutron_veto']['active'] == 'false':
-            if latest_status['neutron_veto']['status'] in active_states:
-                self.StopDetectorGently(detector='neutron_veto')
-            elif latest_status['neutron_veto']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('neutron_veto')
+        active_states = [STATUS.RUNNING, STATUS.ARMED, STATUS.ARMING, STATUS.UNKNOWN]
 
+        for det in latest_status.keys():
+            # The detector should be INACTIVE
+            if goal_state[det]['active'] == 'false':
+                # The detector is not in IDLE, ERROR or TIMEOUT: it needs to be stopped
+                if latest_status[det]['status'] in active_states:
+                    # Check before if the status is UNKNOWN and it is maybe timing out
+                    if latest_status[det]['status'] == STATUS.UNKNOWN:
+                        self.log.info(f"The status of {det} is unknown, check timeouts")
+                        self.log.debug(f"Checking the {det} timeouts")
+                        self.CheckTimeouts(detector=det)
+                    # Otherwise stop the detector
+                    else:
+                        self.log.info(f"Sending stop command to {det}")
+                        self.StopDetectorGently(detector=det)
+               # Deal separately with the TIMEOUT and ERROR statuses, by stopping the detector if needed
```

Comment:
extremely minor:
indentation is wrong, you can add one more space before `# Deal..`